### PR TITLE
Workflow tool copy tweaks

### PR DESCRIPTION
**Pull request by dannon**
xref #5087 

This refines the copy/paste behavior binding of ctrl+c/v to still allow regular use (for example, in tool form or annotation boxes).

It also forces an input blur when nodes are changed.  Kinda surprised this hasn't been an issue before now.

Includes minor cleanup/refactoring.

### CODE REVIEW

File: client/galaxy/scripts/mvc/workflow/workflow-canvas.js [hunk 1]
```diff
@@ -99,24 +99,39 @@ $.extend(CanvasManager.prototype, {
...
-                nodeId = JSON.parse(e.clipboardData.getData('application/json')).nodeId;
-            } catch (error) {}
-            if (nodeId && this.app.workflow.nodes.hasOwnProperty(nodeId)) {
-                this.app.workflow.nodes[nodeId].clone();
+        document.addEventListener("paste", e => {
+            // If it appears that the user is trying to paste into a text box,
+            // pass that through and skip the workflow copy/paste logic.
+            if (
+                document.activeElement &&
+                document.activeElement.type !== "textarea" &&
```
Comment (martenson) [inline client/galaxy/scripts/mvc/workflow/workflow-canvas.js:126]: I like this very much.

**Comment by anatskiy** [comment]
@dannon thanks for the improvements!

**Comment by anatskiy** [comment]
@dannon I notices that if there is just one node on the canvas, it's impossible to copy it, focus on an input element and then paste it because the input is still focused (I assume it's intended behavior). Would it be a good idea to unfocus an input on canvas click too?

**Comment by dannon** [comment]
@anatskiy Hrmm.  Yeah, intended behavior, my main concern is just not breaking the expected copy/paste functionality in the normal text boxes.  I hadn't considered a single node not being able to blur.  I'll blur on the canvas click, too.  (done in https://github.com/galaxyproject/galaxy/pull/5124/commits/0096a5851cd3513e3674341f89f556f710758d20)

**Comment by bgruening** [comment]
Nice! Tested and works! Thanks for improving!

**Pull request merged**